plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/Ros.xtext

```diff
@@ -6,221 +6,282 @@ import "http://www.ipa.fraunhofer.de/primitives" as primitives
+	;
 
+///////////////////
+// YAML format
+///////////////////
+terminal BEGIN: 'synthetic:BEGIN';
+terminal END: 'synthetic:END';
 
-Namespace returns Namespace:
-	GlobalNamespace | RelativeNamespace_Impl | PrivateNamespace;
-
-Package_Impl returns Package:
-	{Package}
-	'Package'
-	name=EString
-	'{'
-		('FromGitRepo' fromGitRepo=EString)?
-		('Specs'  '{' spec+=SpecBase ( "," spec+=SpecBase)* '}')?
-		(artifact+=Artifact ( "," artifact+=Artifact)*)?
-	'}';
+@Override 
+terminal SL_COMMENT: '#' !('\n'|'\r')*;
 
+///////////////////
+// CONVENTIONS AND NAMES
+///////////////////
 EString returns ecore::EString:
 	STRING | ID;
 	
 RosNames returns ecore::EString:
 	ROS_CONVENTION_A | ID | 'node'
 ;
 
-RosParamNames returns ecore::EString:
-	ROS_CONVENTION_PARAM | ID 
-;
-
 terminal ROS_CONVENTION_A:
 	( ('/' ID ) | ( ID '/' ) )* ;
 
 terminal ROS_CONVENTION_PARAM:
 	( ('/' STRING ) | ( STRING '/' ) | ('~' STRING ) )* ;
-	
-Artifact returns Artifact:
-	{Artifact}
-	'Artifact'
-	name=RosNames
-	'{'
-		(node=Node)?
-	'}';
+
+///////////////////
+//PACKAGES
+///////////////////
+Package returns Package:
+	Package_Impl //|
+	//CatkinPackage | 
+	//AmentPackage
+	;
+
+Package_Impl returns Package:
+	{Package}
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('specs:' 
+			BEGIN
+			spec+=SpecBase*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
 
 CatkinPackage returns CatkinPackage:
 	{CatkinPackage}
-	'CatkinPackage'
-	name=RosNames
-	'{'
-		('FromGitRepo' fromGitRepo=EString)?
-		('Dependencies' '{' dependency+=Dependency ( "," dependency+=Dependency)* '}' )?		
-		('Specs'  '{' spec+=SpecBase ( "," spec+=SpecBase)* '}')?
-		(artifact+=Artifact ( "," artifact+=Artifact)*)?
-	'}';
-	
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('artifacts:' 
+			BEGIN 
+			artifact+=Artifact*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
+
 AmentPackage returns AmentPackage:
 	{AmentPackage}
 	'AmentPackage'
-	name=RosNames
-	'{'
-		('FromGitRepo' fromGitRepo=EString)?
-		('Dependencies' '{' dependency+=Dependency ( "," dependency+=Dependency)* '}' )?		
-		('Specs'  '{' spec+=SpecBase ( "," spec+=SpecBase)* '}')?
-		(artifact+=Artifact ( "," artifact+=Artifact)*)?
-	'}';
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('artifacts:' 
+			BEGIN 
+			artifact+=Artifact*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
+
+///////////////////
+//ARTIFACT AND NODE
+///////////////////
+Artifact returns Artifact:
+	{Artifact}
+		name=RosNames':'
+		BEGIN
+		(node=Node)?
+		END;
 
-ServiceSpec returns ServiceSpec:
-	{ServiceSpec}
-	'ServiceSpec'
-	name=EString
-	'{'
-		('request' request=MessageDefinition)?
-		('response' response=MessageDefinition)?
-	'}';
+Node returns Node:
+	'node:' name=RosNames
+	 BEGIN
+		(
+		('publishers:' 
```

Comment:
One option is to a dd name tag. This will force it to be directly after "node:". 
```suggestion
Node returns Node:
	'node:'
	 BEGIN
	        'name:' name=RosNames
		(
		('publishers:' 
```
This seems to also be a problem in other places. We should go with a general solution. I think adding a name tag is probably the easiest and most general solution.